// Copyright 2022 PingCAP, Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.#pragma once

syntax = "proto2";

package dtpb;

message PackProperty {
    // when gc_safe_point exceed this version, there must be some data obsolete in this pack
    required uint64 gc_hint_version = 1;
    // effective rows(multiple versions of one row is count as one include delete)
    required uint64 num_rows = 2;
    // the number of rows in this pack which are deleted
    optional uint64 deleted_rows = 3;
}

message PackProperties {
    repeated PackProperty property = 1;
}

message EmbeddedChecksum {
    required string name = 1;
    required bytes checksum = 2;
}

message ChecksumDebugInfo {
    required string name = 1;
    required string content = 2;
}

message ChecksumConfig {
    required uint64 checksum_algorithm = 1;
    required uint64 checksum_frame_length = 2;
    repeated EmbeddedChecksum embedded_checksum = 3;
    // checksum of the above field
    required bytes data_field_checksum = 4;
    // additional information
    repeated ChecksumDebugInfo debug_info = 5;
}
